### DISCUSSION TITLE
CreateProfileView 질문

### DISCUSSION THREAD

**Discussion by alice0047** [ttakkeun/Front_iOS#62] [Q&A]
1. 각 필드가 입력이 되면, 체크 표시가 되어야 함
<img width="347" alt="스크린샷 2024-08-01 오후 12 42 29" src="https://github.com/user-attachments/assets/23c6ecb8-68cb-4076-8580-c1734441fd6e">
그럼 선택된 버튼이나 입력된 값이 추적되어야 하는데, View에서 @State 변수를 작성해서 사용자가 입력한 각 값을 저장한 뒤 사용하는 것이 맞나..?
<img width="372" alt="스크린샷 2024-08-01 오후 12 45 15" src="https://github.com/user-attachments/assets/449f37a8-fbcd-463e-a491-a715a366bca0">

2. 위에가 맞다면, 등록하기 버튼을 눌렀을 때, 한 번에 해당 값들을 넘겨줘야 할텐데, 이름, 종, 품종, 생년월일, 중성화여부를 CreateProfileViewModel()라는 Struct에 하나로 담아서 API를 call하면 되는게 맞을까? 이걸 View에서 등록하기 버튼의 액션으로 주면 되는 것인지!

3. 버튼은 예를 들어 '강아지' 버튼을 클릭하면 해당 필드에는 강아지가 저장되어야하는데, API 명세서에는 String으로 되어 있음. 그러면 해당 버튼이 눌리면 위에 정의한 State 변수에 'DOG' String을 저장하면 되는게 맞겠죠..?

3. ViewModel 코드를 작성하고 있는데, Post와 관련된 스승님의 코드가 Login밖에 없어서 적용에 어려움을 겪고 있음. 혼자 공부해보고 해결해보도록 하겠음.

**Comment by alice0047** [comment]
+  해당 필드 작성되었는지 확인하기 위해서 아래와 같은 Bool type을 사용하라고 지피티가 알려주는데 괜찮겠죠..?
    
    @State private var isNameFilled: Bool = false
    @State private var isTypeSelected: Bool = false
    @State private var isVarietyFilled: Bool = false
    @State private var isBirthFilled: Bool = false
    @State private var isNeutralizationSelected: Bool = false

**Reply by JEONG-J** [reply]
좋아
1번 : 아니요!
2번 : 틀리니 답하지 않겠음
3번 : 1번을 알면 해결됨
4번 : 너무 추상적임 어려움이라는게 뭘 말하는건지 알려줘야 도와주지 어렵다는게 본인이 느끼기에 어느쪽인지 나는 모름

**Reply by JEONG-J** [reply]
자 이제 지피티가 알려준거에 대해 말하도록 해보지..

참거짓으로 뒀다는건 어떤 기준을 가지고 참이냐 거짓이냐 되도록 했다는건데 지금 그 기준도 못잡은거 같은데 무작정 왜 참거짓을 꺼내든거지?? 코드를 그대로 쓸거면 너가 만들지 말고 지피티한테 피그마 사진 보여주고 다 해달라하지 지금 뭐하는 행동이지??

코드를 보고 지피티의 타당성을 좀 생각해봐라 아니 진짜.. 라뗴는 말이다.. 모르면 지피티 없을 때 모르면 진짜 다 뒤져가며 따지며 공부하고 그랬는데 진짜 왜 지피티만 보면 이게 정답이다라고 생각하고 덤비는거야?? 이거 걍 진짜 지피티 나온 이래로 학생들 보면 죄다 그럼 공부가 아니라 코드를 대타로 맡기는거랑 뭐가 다름
